File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Core/TeleComSystem.java
package com.github.grzesiek_galezowski.BastardInjection.Core;

import com.github.grzesiek_galezowski.BastardInjection.Inbound.BinaryUdpInbound;
import com.github.grzesiek_galezowski.BastardInjection.Inbound.IInbound;
import com.github.grzesiek_galezowski.BastardInjection.Outbound.IOutbound;
import com.github.grzesiek_galezowski.BastardInjection.Outbound.Outbound;

import java.io.Closeable;
import java.io.IOException;

public class TeleComSystem implements Closeable {
  private final IProcessingWorkflow _processingWorkflow;
  private final IInbound _inbound;
  private final IOutbound _outbound;

  public TeleComSystem() {
    this(new AcmeProcessingWorkflow(), new BinaryUdpInbound(), new Outbound());
  }

  //for tests
  public TeleComSystem(
      IProcessingWorkflow processingWorkflow,
      IInbound inbound,
      IOutbound outbound) {
    _processingWorkflow = processingWorkflow;
    _inbound = inbound;
    _outbound = outbound;
  }

  public void start() {
    _inbound.setDomainLogic(_processingWorkflow);
    _processingWorkflow.setOutbound(_outbound);
    _inbound.startListening();
  }

  public void close() throws IOException {
    _inbound.close();
  }
}

File: CSharp/DependencyInjection/DependencyInjection/BastardInjection/_4_BastardInjection/Core/TeleComSystem.cs
using System;
using BastardInjection._4_BastardInjection.Inbound;
using BastardInjection._4_BastardInjection.Outbound;

namespace BastardInjection._4_BastardInjection.Core
{
  class TeleComSystem : IDisposable
  {
    private readonly IProcessingWorkflow _processingWorkflow;
    private readonly IInbound _inbound;
    private readonly IOutbound _outbound;

    public TeleComSystem()
      : this(new AcmeProcessingWorkflow(), new BinaryUdpInbound(), new Outbound.Outbound())
    {
      
    }

    //for tests
    public TeleComSystem(
      IProcessingWorkflow processingWorkflow,
      IInbound inbound,
      IOutbound outbound)
    {
      _processingWorkflow = processingWorkflow;
      _inbound = inbound;
      _outbound = outbound;
    }

    public void Start()
    {
      _inbound.SetDomainLogic(_processingWorkflow);
      _processingWorkflow.SetOutbound(_outbound);
      _inbound.StartListening();
    }

    public void Dispose()
    {
      _inbound.Dispose();
    }
  }
}